grafana-plugin/src/containers/UserSettings/parts/tabs/GoogleCalendar/GoogleCalendar.tsx

```diff
@@ -43,111 +41,141 @@ const GoogleCalendar: React.FC<{ id: ApiSchemas['User']['pk'] }> = observer(({ i
 
-  const user = userStore.items[id];
-
   return (
-    <VerticalGroup>
-      <Block bordered className={styles.root}>
-        <VerticalGroup spacing="lg">
-          {user.has_google_oauth2_connected ? (
-            <VerticalGroup>
-              <HorizontalGroup justify="space-between">
-                <HorizontalGroup>
-                  <GoogleCalendarLogo width={32} height={32} />
-                  <Text>Google calendar is connected</Text>
-                </HorizontalGroup>
-                <WithPermissionControlTooltip userAction={UserActions.UserSettingsWrite}>
-                  <WithConfirm title="Are you sure to disconnect your Google account?" confirmText="Disconnect">
-                    <Button variant="destructive" onClick={userStore.disconnectGoogle}>
-                      Disconnect
-                    </Button>
-                  </WithConfirm>
-                </WithPermissionControlTooltip>
-              </HorizontalGroup>
-            </VerticalGroup>
-          ) : (
-            <HorizontalGroup justify="space-between" spacing="lg">
-              <HorizontalGroup spacing="md">
-                <GoogleCalendarLogo width={32} height={32} />
-                <div>
-                  <Text.Title level={5}>Connect your Google Calendar</Text.Title>
-                  <Text type="secondary">
-                    This connection allows Grafana OnCall to read your Out of Office events and autogenerate Shift Swap
-                    Requests
-                  </Text>
-                </div>
-              </HorizontalGroup>
+    <Block bordered className={utils.width100}>
+      <VerticalGroup spacing="lg">
+        {user.has_google_oauth2_connected ? (
+          <VerticalGroup>
+            <HorizontalGroup justify="space-between" spacing="lg" align="flex-start">
+              <Heading connected />
               <WithPermissionControlTooltip userAction={UserActions.UserSettingsWrite}>
-                <Button variant="primary" onClick={UserHelper.handleConnectGoogle}>
-                  Connect
-                </Button>
+                <WithConfirm title="Are you sure to disconnect your Google account?" confirmText="Disconnect">
+                  <Button variant="destructive" onClick={userStore.disconnectGoogle}>
+                    Disconnect
+                  </Button>
+                </WithConfirm>
               </WithPermissionControlTooltip>
             </HorizontalGroup>
-          )}
+          </VerticalGroup>
+        ) : (
+          <HorizontalGroup justify="space-between" spacing="lg" align="flex-start">
+            <Heading connected={false} />
+            <WithPermissionControlTooltip userAction={UserActions.UserSettingsWrite}>
+              <Button variant="primary" onClick={UserHelper.handleConnectGoogle}>
+                Connect
+              </Button>
+            </WithPermissionControlTooltip>
+          </HorizontalGroup>
+        )}
+
+        {user.has_google_oauth2_connected && (
```

Comment:
There's a helper `RenderConditionally` for such conditionals

Reply:
ok, I'll keep that in mind for the future